### PR TITLE
Issue 33: Added code examples to api-documentation

### PR DESCRIPTION
**Pull request by kstigen**

### CODE REVIEW

File: weather_api/__main__.py [hunk 1]
```diff
@@ -1,19 +1,28 @@
...
-from weather_api.routes import system_resources, weather_routes
 from weather_api.config import settings
+from weather_api.openapi import openapi
+from weather_api.routes import system_resources, weather_routes
 
 
 def get_application() -> FastAPI:
+    this_dir = pathlib.Path(__file__).parent
+
     api = FastAPI(
```
Comment (dymbe) [inline weather_api/__main__.py:13]:
```python
If you want the doc to have a favicon too you need to create your own `/redoc`-endpoint:

app = FastAPI(
    lifespan=lifespan,
    root_path=settings.api_root_path,
    redoc_url=None,
)


@app.get("/redoc", include_in_schema=False)
def redoc():
    return get_redoc_html(
        openapi_url="/openapi.json",
        title="Geocoder API",
        redoc_favicon_url="https://www.openepi.io/favicon.ico",
    )

**Review by dymbe — CHANGES_REQUESTED** [review]
Looks good! I did some things a bit differently in my PR, you should take a look at it so we can agree on how we want to do this on all APIs
https://github.com/openearthplatforminitiative/geocoder-api/pull/10/files

File: weather_api/openapi/openapi.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            if os.path.isfile(file_with_code_sample):
+                with open(file_with_code_sample) as f:
+                    code_template = Template(f.read())
+                    code_samples.append(
+                        {
+                            "lang": lang,
+                            "source": code_template.safe_substitute(
+                                api_host=settings.public_url,
+                                api_root_path=settings.api_root_path,
+                                api_endpoint=route.path,
```
Comment (dymbe) [inline weather_api/openapi/openapi.py:50]: Maybe the files can be made a bit simpler if you concatenate the URL here, and only pass in a single variable?

File: weather_api/config.py [hunk 3]
```diff
@@ -3,13 +3,22 @@
...
-    api_root_path: str = "/"
+    api_root_path: str = ""
 
     met_api_url: str = "https://api.met.no/weatherapi"
     api_description: str = 'This is a RESTful service that provides accurate and up-to-date weather information based on data sourced from <a href="https://api.met.no">https://api.met.no</a>. <br/>The data are freely available for use under a <a href="https://api.met.no/doc/License">Creative Commons license</a>.'
 
+    api_domain: str = "localhost"
+
+    @property
+    def public_url(self):
```
[the remark this answers is not in the document]
Comment (kstigen) [reply weather_api/config.py:17]: Can call it exposed_url or something like that

**Pull request merged**